### DISCUSSION TITLE
Failed to scrape resource collection 'Application-insight-landscape' - resource discovery group

### DISCUSSION THREAD

**Discussion by mgLearningAccount2** [tomkerkhove/promitor#2066] [Q&A] [answered]
I have made a simple configuration setup with Prometheus -- Promitor -- Azure AppInsight , these config are done locally without any assist from kubernetes. 

Setup Details:
--------------
docker run command:
docker run -d -p 8999:80 --name promitor-agent-scraper --env AUTH_APPKEY=xxxxxx-xxxx-xxxx-xxxxxx --volume c:/promitor/metrics-declaration.yaml:/config/metrics-declaration.yaml --volume c:/promitor/runtime.yaml:/config/runtime.yaml ghcr.io/tomkerkhove/promitor-agent-scraper:2.6.0

![image](https://user-images.githubusercontent.com/106647650/176686786-d4d230ad-cd9f-4e80-bcaa-0f91f01e3e6b.png)

![image](https://user-images.githubusercontent.com/106647650/176687015-ecd12495-51cc-493c-8d85-aaf622deae06.png)

![image](https://user-images.githubusercontent.com/106647650/176687097-60856ee6-e600-4ea3-b828-52087cc5e242.png)

![image](https://user-images.githubusercontent.com/106647650/176734240-6430ee5d-06c9-41a8-9fc5-eef833ab7c50.png)

Currently my Promitor docker is running successfully and their localhost/metric endpoint are available Prometheus target. But, I am facing few issues between Promitor and azure configuration. 

1. how to expose an appinsight data metric to metric-declaration.yml? I have updated the metric.yml file with azure appinsight resources is this the resource group we have to refer? and does the metric only available when data is present? 

![image](https://user-images.githubusercontent.com/106647650/176733974-a498969e-9106-489f-931e-1ec7cf5616ba.png)

And in the AzureMetricConfiguration should I be using namespace or the metric value from AppInSight.
![image](https://user-images.githubusercontent.com/106647650/176733777-77579f79-0319-4aed-98c8-eb5b29c33410.png)

2. The another issue I was facing is the "resource discovery group", based on [doc](https://docs.promitor.io/v2.6/how-it-works/#scraping-azure-monitor-metrics) Promitor discovery agent query the data from azure resource graph. Even thought these config are mentioned as optional, I am getting error on this when the container is running. Is this to be ignored or if left will it affect data scrapping.

![image](https://user-images.githubusercontent.com/106647650/176806182-8729b79c-0aa8-4cf1-bc92-1bb98ce5dcc0.png)
![image](https://user-images.githubusercontent.com/106647650/176689350-08771b01-a9d1-470e-8306-afb2d44ef413.png)

4. In runtime.yaml configuring the Authentication part for mode & indentityId (where these value are taken from exactly), is this required and under which condition to be included? 
error when running the container
![image](https://user-images.githubusercontent.com/106647650/176731535-0448c11c-f78a-455b-b431-d8a9b8a5b1a8.png)

**Comment by tomkerkhove** [comment]
> 1. how to expose an appinsight data metric to metric-declaration.yml? I have updated the metric.yml file with azure appinsight resources is this the resource group we have to refer? and does the metric only available when data is present?

Sorry, I'm not sure I get your question here?

> 2. The another issue I was facing is the "resource discovery group", based on [doc](https://docs.promitor.io/v2.6/how-it-works/#scraping-azure-monitor-metrics) Promitor discovery agent query the data from azure resource graph. Even thought these config are mentioned as optional, I am getting error on this when the container is running. Is this to be ignored or if left will it affect data scrapping.

The URI that you have configured is not reachable and should be the DNS name of your container. Make sure you are running an instance of the resource discovery agent as well and that the scraper is able to access it.

> 4. In runtime.yaml configuring the Authentication part for mode & indentityId (where these value are taken from exactly), is this required and under which condition to be included?

Yes this is required to authenticate with Microsoft Azure. It should be an Azure AD application for which you'll need the application ID & secret.

**Reply by mgLearningAccount2** [reply]
Thanks for your reply..
Before I dive deep into these three questions briefly, please help me to figure out what are the necessary steps (are mandatory) things that should I be considering from Azure and Promitor end for the configuration like a pre-requisites steps. And also while configuring the runtime and metric-declaration.yml file what are the values should I be passing it from the azure services to Promitor and vise-versa.

**Reply by tomkerkhove** [reply]
I think that's all clearly covered in the documentation, no? I'm happy to help but that question is a bit broad; can you be more concrete please?

This should give you an indication: https://docs.promitor.io/v2.7/walkthroughs/scrape-promitor-with-prometheus-on-azure-kubernetes-service/#introduction

It's aimed at Kubernetes but the general setup of creating identity and such is correct though.

**Reply by mgLearningAccount2** [reply]
I have made few changes to the metric declaration file and facing following issues. These configuration are done locally without any use of AKS( as I don't have any access for them and trying see if the data flow can be achieved without any need for Kubernetes).

 Auth mode: service provide (for the App registration) are given with read access on the AppInsight resource only. My Promitor localhost where picked up by Prometheus target  the problem is around the Azure and promitor part
version: 2.7.0

Issue 1:
--------
![image](https://user-images.githubusercontent.com/106647650/177546714-d7c4d57f-f139-4d57-bc34-76c2d152e359.png)

Issue 2:
--------
![image](https://user-images.githubusercontent.com/106647650/177547186-09491854-52c6-4d9a-9541-6bd1c521c2d6.png)

**Reply by tomkerkhove** [reply]
Can you post your full configuration and logs please?

**Reply by mgLearningAccount2** [reply]
@tomkerkhove thanks for the reply. I was able to run it now, the issue was the Authentication type I was using. Even though I am getting some data over the Prometheus, there are few fail info 

![image](https://user-images.githubusercontent.com/106647650/187039218-858aa5dd-4200-4ccb-b2a0-665a752af658.png)
config file for metric:
![image](https://user-images.githubusercontent.com/106647650/187039657-56317631-2dd5-4403-9bc5-14834f2e045c.png)

**Reply by tomkerkhove** [reply]
This is related to #2089

**Comment by mgLearningAccount2** [comment, marked as answer]
Thanks @tomkerkhove closing this discussion and will follow up on the https://github.com/tomkerkhove/promitor/issues/2089 issue for updates.

Marking https://github.com/tomkerkhove/promitor/discussions/2066#discussioncomment-3490134 as Answered
